### DISCUSSION TITLE
Enabling PKCS#12 certificate for librdkafka usage (RC2-40-CBC algorithm)

### DISCUSSION THREAD

**Discussion by csolamx** [edenhill/librdkafka#4385] [Q&A] [answered]
Hello,

I am trying to enable secure communication with Kafka from a MS .NET project.  I am using a PKCS#12 certificate, however when I attempt communication I get the following exception:

**Failed to parse ssl.keystore.location PKCS#12 file: cert.pfx: error: 0308010C:digital envelope routines::unsupported: Global default library context, Algorithm (RC2-40-CBC : 0), Properties ()**

I am aware that this might have to do with the certificate implementing a legacy encryption algorithm, however I might have to use it as it is.  Is there some way to enable this certificate with librdkafka?  If not, what's the best way to fix it to make it usable with the library?

Regards

**Comment by pranavrth** [comment, marked as answer]
Use `ssl.providers:legacy,default` to use legacy algorithms as well with the default provider in librdkafka. You can find more information [here](https://docs.confluent.io/platform/current/clients/librdkafka/html/md_CONFIGURATION.html)

**Reply by csolamx** [reply]
Thanks a lot!

**Comment by xfiodembo** [comment]
this is broken btw in 2.3.0 
https://github.com/confluentinc/confluent-kafka-python/issues/1596
